baybe/recommenders/pure/bayesian/sequential_greedy.py

```diff
@@ -79,7 +79,7 @@ def _recommend_discrete(
         except AttributeError as ex:
             raise NoMCAcquisitionFunctionError(
                 f"The '{self.__class__.__name__}' only works with Monte Carlo "
-                f"acquisition functions."
+                f"acquisition functions for batch sizes > 1."
```

Comment:
I think it actually makes sense to remove the `try-catch` and instead raise this error in a guard clause at the top under the appropriate condition (i.e. `batch_size>1` and non-MC function). Because I've already had weird situations several times that a **different** AttributeError was thrown (while developing) and this got me completely off track. And I don't see a reason really why should wait for the optimization to fail... Right?

Reply:
yes this atttribute error ctach ehre can have several rasons and  ive also encountered it for some of the (not included) new acqf functions that complain about something else than our error here. It def needs to be fixed

Reply:
having said that the batch size change you suggest would not be enough for that

Reply:
What you you mean with `batch size change`? My suggestion was simply to have an if-condition that checks for `batch_size>1` AND `not is_mc` ...

Reply:
yes i could do that but it wont solve any of the other issues that could cause different attribute errors

Reply:
sure, if there is more unfixed stuff, this will propagate through. but at least we then see the correct error, and not a completely unrelated error message 👍🏼

Reply:
added something along those line

Reply:
Honestly, I think the version you have now is overly complicated!? Why not simply remove the entire try-except block altogether and move your if-condition to the very top of the method, even in front of the costly `botorch` import? You don't need to wait for the optimizer to throw the error just to then catch it, because we *know* it will the error in that case. And then the remaining exceptions are just directly raised, without being caught in `try` first.

Reply:
good point I changed it accordingly

makes me wonder tho how are we supposed to deal with the now implicitly changed docstring? Its now got an additional raise which should be mentioned in the raises section. But previously this function just refers to the base class via `# See base class.`. Should I repeat the entire docstring of that or just leave it out? Maybe also for @AVHopp

Reply:
Thx! Docstring and exception handling is anyway a nightmare in Python... So let's see if Alex has some clever ideas, but I'm against copy-paste, as you probably could have already imagined 😄

Reply:
Well, if the function has a different functionality/things are different from the original version, it deserves a new docstring, even if it is 90% the same imo. This is still better than having an incorrect docstring

Reply:
Ok, then let @Scienfitz add something, but I recommend: let's use the words "override/extend" to mention only differences/extension compared to the base class one (that would be a compromise I think)

Reply:
Yes, and we can just explicitly link to the original function and then add stuff if that is fine with our linters

Reply:
I propose the following 
```python
        """Generate recommendations from a discrete search space.

        Note: This method overwrites
        :func:`baybe.recommender.pure.base.PureRecommender._recommend_discrete`
         keeping the same characteristics except the ones mentioned here.

        Raises:
            NoMCAcquisitionFunctionError: If a non-Monte Carlo acquisition function is
                used with a batch size > 1.
        """
```

its part of this PR via amending the last commit
keep in mind this is wont shown in the doc anyway because its a private function
please let me know if that can go ahead

I notice also inconsistent behavior here, in the same file we have `_recommend_hybrid` which also overwrites the base class implementation, but has its own completely copypasta docstring with an amended raises section

Reply:
never mind its not possible to do that without violating lots of stuff
![image](https://github.com/emdgroup/baybe/assets/17951239/f6e0b1f8-1788-47e1-82f7-7e00437448ab)

I would simply follow the approach of `_recommend_hybrid`

Reply:
and the same problems were also part of conti and hyprid error raising so I changed it there too
Seems our previous workflow was to copy the entire docstirng+amend which I did for those as well

Reply:
Ok, I'm not a happy guy but I think there is simply no "good" approach, so I'll simply look away 🙈